src/backend/distributed/commands/utility_hook.c

```diff
@@ -855,7 +866,7 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		 * Since we must have objects on workers before distributing them,
 		 * mark object distributed as the last step.
 		 */
-		if (ops && ops->markDistributed)
+		if (ops && ops->markDistributed && shouldExecutePropagationSteps)
```

Comment:
Similarly. why do we need this check in post-Postprocess? If the address is not valid, PG should have already errored out?